File: rnd.py
import random as ran
from pynput import mouse
import time as t
import keyboard as kb
moves = ["up", "down", "left", "right", "v", "s", "a", "b", "L", "R"]
rand = ran.choice(moves)
count = 0

kb.wait("k")
go = True

start_time = t.time()  # Record the start time

def on_click(x, y, button, pressed):
    global go
    if not pressed and button == mouse.Button.right:
        go = False
        print(count)
        print("Program was running for:", t.strftime("%H:%M:%S", t.gmtime(t.time() - start_time)))

# Create a listener for mouse clicks
listener = mouse.Listener(on_click=on_click)
listener.start()

while go:
    rand = ran.choice(moves)
    kb.press(rand) 
    print(rand) # Just to see the random move, you can remove this in actual implementation
    count += 1
    
    # Perform your action here
    # kb.press(rand) # You can include this to simulate keyboard press
    # Assuming you want to end when 'i' key is pressed
    # if kb.is_pressed("i"):
    #     print(count)
    #     break

# Stop the listener after loop ends
listener.stop()

File: rnd.lua
local random = math.random
local os_time = os.time
local kb = require("keyboard")
local mouse = require("mouse")

local moves = {"up", "down", "left", "right", "v", "s", "a", "b", "L", "R"}
local count = 0
local go = true

local start_time = os_time()  -- Record the start time

function on_click(x, y, button, pressed)
    if not pressed and button == 2 then  -- Button 2 represents the right mouse button
        go = false
        print("Number of moves:", count)
        print("Program was running for:", os.date("!%X", os_time() - start_time))  -- Print duration
    end
end

-- Register the mouse listener
mouse.on("click", on_click)

-- Main loop
while go do
    local rand = moves[random(#moves)]  -- Choose a random move from the list
    kb.press(rand)
    print(rand)  -- Just to see the random move, you can remove this in actual implementation
    count = count + 1

    -- Perform your action here
    -- kb.press(rand)  -- You can include this to simulate keyboard press

     -- Wait for 0.1 seconds between each move
end
